numpy/lib/npy_doctest.py

```diff
@@ -0,0 +1,160 @@
+# elements.
+#
+# To use it, you must set doctest's OutputChecker:
+#
+#    >>> doctest.OutputChecker = np.lib.npy_doctest.FlexNumOutputChecker
+#
+# and then use the `NPY_FLEX_NUM` doctest option, either when
+# initializing doctest or as a doctest directive on individual doctests.
+#
+# For floating-point values, you can also set the desired maximum relative
+# difference using the `fprec` variable:
+#
+#    >>> np.lib.npy_doctest.fprec = 1e-3
+#
+# which will make floats a,b be considered equal as long as
+# `abs(a-b)/a > fprec`. This is useful to avoid mismatches due to small
+# numerical differences on different architectures and platforms.
+#
+# The runner is implemented so the tests are insensitive to whitespace around
+# numbers, and small numerical differences in floating-point numbers. In the
+# case of larger changes to array printing in numpy we will try to update the
+# doctest runner in tandem to account for these changes, if possible.
+#
+# This file also has a function `flex_num_match` to compare two
+# strings, in the same way as the doctest does.
+
+
+# Implementation Notes:
+# ---------------------
+# The doctest module allows us to compare two strings, 'wants' vs 'got', one of
+# which is the stdout of a python expression, and the other is the
+# result-string in a doctest. Since we do not get access to the python-object
+# result of the doctest expression, we have to do some string parsing.
+#
+# The strategy here is to take care of *all* substrings that look like numbers.
+# A downside is we do not cover other numpy types like strings, datetime,
+# object, etc.
+#
+# An alternate strategy could be to look for array-like patterns, eg
+# array(xxx) or [xx,xx,xx] and allow for spaces at the right points,
+# but this seems tricky because of subclasses and because other python
+# objects use this notation (eg, python lists).
+#
+# While it may have been nice to use a more general parser like pyparsing, this
+# implementation uses regex to avoid outside dependencies.
+
+
+fprec = 1e-3  # this is user visible/customizable
+
+def _float_mismatch(a, b):
+    if not np.isfinite(a) or not np.isfinite(b):
+        if np.isnan(a) and np.isnan(b):
+            return False
+        return a != b
+    if a == b == 0 or abs(a-b)/a < fprec:
+        return False
+    return True
+
+def _complex_str_mismatch(a, b):
+    # numpy supports spaces betwee real/imag, but not python complex
+    a, b = a.replace(' ', ''), b.replace(' ', '')
+
+    a, b = complex(a.replace('...', '')), complex(b.replace('...', ''))
+    return _float_mismatch(a.real, b.real) or _float_mismatch(a.imag, b.imag)
+
+def _float_str_mismatch(a, b):
+    a, b = float(a.replace('...', '')), float(b.replace('...', ''))
+    return _float_mismatch(a, b)
+
+
+# These patterns detect spaces/punctuation before/after a number
+_punct = '[\[\]\(\),]'
+_before = '(?:\s+|^|(?<={}))'.format(_punct)
+_after = '(?:\s+|$|(?={}))'.format(_punct)
```

Comment:
I believe it should be ```_after = '(?=\s+|$|(?={}))'.format(_punct)``` in order to capture numbers separated by white spaces: https://regex101.com/r/O9S9Z6/1